libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -471,72 +480,15 @@ internal class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
-        unconditionalStatusPollingJob?.cancel()
         if (state != TripSessionState.STARTED) return
 
         this.rawLocation = rawLocation
         locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         mainJobController.scope.launch {
             navigator.updateLocation(rawLocation.toFixLocation())
```

Comment:
Could you check @SiarheiFedartsou @DmitryAk when you have a chance? 🙏

Reply:
What do you mean by "blocking call" here? If you mean if it can block current thread for significant amount of time then no, I don't think so.

Reply:
What do you consider as `significant amount of time`? If it's locked somehow internally we should probably keep switching threads on our side.

Reply:
Corrected answer: we don't perform any IO or "hard" algorithms inside `updateLocation`, but at the moment it is synchronized with internal "getStatus" calls(which is performed by us, but not by SDK after transition to callback approach) and that's why it can block for some amount of time.
@DmitryAk please confirm when you have a chance

Reply:
@SiarheiFedartsou absolutely agree: I'm worried about this lock: https://github.com/mapbox/mapbox-navigation-native/blob/1b2c53d49614e4ffe1b498699f9ed06926b95829/src/navigator_impl.cpp#L456 - the same is used on `getStatus` calculation - so the `updateLocation` can be blocked for a long time.
We have an issue about this lock usage: https://github.com/mapbox/mapbox-navigation-native/issues/2308
So this `updateLocation` call seems blocking currently.